cpp/src/strings/convert/convert_fixed_point.cu

```diff
@@ -0,0 +1,450 @@
+
+    DecimalType value  = 0;      // used for overflow checking
+    bool decimal_found = false;  // mainly for checking duplicate decimal points
+    int32_t curr_scale = scale;  // running scale for scale < 0 case
+    while (iter != iter_end) {   // check all bytes for valid characters
+      auto const chr = *iter++;
+      if (chr == '.' && !decimal_found) {
+        decimal_found = true;
+        continue;
+      }
+      if (chr < '0' || chr > '9') return false;            // invalid character check
+      if (iter > iter_check && curr_scale >= 0) continue;  // overflow checking no longer needed
+
+      // check for overflow in the integer component
+      auto const digit     = static_cast<DecimalType>(chr - '0');
+      auto const max_check = (std::numeric_limits<DecimalType>::max() - digit) / DecimalType{10};
+      if (value > max_check) return false;
+      value = (value * DecimalType{10}) + digit;
+
+      // increment running scale if we are right of the decimal point
+      curr_scale += (decimal_found && curr_scale < 0);
+    }
+    // check overflow on any remaining negative scale value
+    if ((curr_scale < 0) &&
+        (value > (std::numeric_limits<DecimalType>::max() /
+                  static_cast<DecimalType>(exp10(static_cast<double>(-curr_scale))))))
+      return false;
+
+    // everything passed
+    return true;
+  }
+};
+
+/**
+ * @brief The dispatch function for converting strings column to fixed-point column.
+ */
+struct dispatch_to_fixed_point_fn {
+  template <typename T, std::enable_if_t<cudf::is_fixed_point<T>()>* = nullptr>
+  std::unique_ptr<column> operator()(strings_column_view const& input,
+                                     data_type output_type,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr) const
+  {
+    using DecimalType = device_storage_type_t<T>;
+
+    auto const d_column = column_device_view::create(input.parent(), stream);
+
+    // create output column
+    auto results   = make_fixed_point_column(output_type,
+                                           input.size(),
+                                           cudf::detail::copy_bitmask(input.parent(), stream, mr),
+                                           input.null_count(),
+                                           stream,
+                                           mr);
+    auto d_results = results->mutable_view().data<DecimalType>();
+
+    // convert strings into decimal values
+    thrust::transform(rmm::exec_policy(stream),
+                      thrust::make_counting_iterator<size_type>(0),
+                      thrust::make_counting_iterator<size_type>(input.size()),
+                      d_results,
+                      string_to_decimal_fn<DecimalType>{*d_column, output_type.scale()});
+    results->set_null_count(input.null_count());
```

Comment:
```suggestion
```
I think this is redundant (null count is passed at creation time)

Reply:
If you look at the source for `column::mutable_column_view()` (which is called above) you will see that it invalidates the `null_count`
https://github.com/rapidsai/cudf/blob/d3f5add210293a4832dafb85f04cbb73149b9d54/cpp/src/column/column.cu#L133
This sets the null-count back to the known correct value to save from it being forced to be recomputed when accessed later.